vivisect/analysis/elf/elfplt_late.py

```diff
@@ -0,0 +1,325 @@
+
+            offset += lsz
+
+
+    # PLT-Func-Distance method
+    if not len(distanceheur):
+        logging.info("skipping analyzePLT(0x%x, 0x%x): no existing functions found", pltva, pltsz)
+
+    else:
+        ######## Now let's attempt to identify the smallest common distance between functions
+        # what's the smallest distance between functions that
+        logger.info("... analysis mode 2: PLT-Func-Distance")
+        curpltcnt = len(curplts)
+        minimumbar = 1 + (curpltcnt // 100)
+
+        # cull the heard.  our winner wants to be at least greater than one entry
+        for delta, count in list(distanceheur.items()):
+            if count < minimumbar:
+                distanceheur.pop(delta)
+
+            elif delta == 0:  # can't have 0.  skip the first PLT function
+                distanceheur.pop(delta)
+
+        workdist = list(distanceheur.items())
+        workdist.sort()
+
+        if not len(workdist):
+            logger.info('... bailing, no existing PLT functions in this section to compare to...')
+
+        else:
+            # first entry should be our guy...
+            funcdist = workdist[0][0]
+            logger.debug("funcdist: 0x%x", funcdist)
+
+            idx = getGoodIndex(curplts, funcdist)
+            fva = curplts[idx]
+
+            ### magic check:  we are working off incomplete data... it's completely possible our
+            # gap measurement is twice as big as it should be (or 3 times).  how to check is 
+            # where the magic comes in...
+            funcsz = vw.getFunctionMeta(fva, 'Size')
+            for divisor in range(funcdist, 0, -1):
+                logger.log(e_cmn.SHITE, 'attempting to divide funcdist (0x%x) by %d', funcdist, divisor)
+
+                # does the gap between functions support splitting?
+                if funcdist < (divisor * funcsz):
+                    logger.log(e_cmn.SHITE, '.. not big enough')
+                    continue
+
+                newdist = funcdist // divisor
+                # does the funcdist split evenly?
+                if newdist * divisor != funcdist:
+                    logger.log(e_cmn.SHITE, ".. doesn't divide evenly(newdist: 0x%x  divisor: 0x%x   funcdist: 0x%x)", newdist, divisor, funcdist)
+                    continue
+
+                # do the opcodes support this size split?
+                if not compareFuncs(vw, fva, fva + newdist, funcsz):
+                    logger.log(e_cmn.SHITE, ".. functions don't match!")
+                    continue
+
+                break
+
+            # should end up dividing by 1 if not divisible
+            if divisor > 1:
+                logger.info("dividing our lowest function (0x%x) distance by %d", funcdist, divisor)
+                funcdist //= divisor
+
+            idx = getGoodIndex(curplts, funcdist)
+
+            tmpva = curplts[idx]
+            logger.debug("... backwards from... 0x%x", tmpva)
```

Comment:
What do you mean by backwards from?

So this is kinda why I'm not a fan of the sheer amount of logging we've got. There's going to be enough other messages at this level where "backwards from" is not going to be particularly helpful as a logging message (even in context)

Reply:
if you see it in context it should make more sense.  
this method identifies a good starting point where we know we have two PLT entries and we've guessed the distance between them.  from there, we move backward toward the beginning of the PLT and then we start again from the same point and go forwards toward the end of the PLT (keep in mind, there are often multiple PLT's which often use different forms).